include/utility.h

```diff
@@ -249,19 +263,19 @@ class ParamServer
         sensor_msgs::Imu imu_out = imu_in;
         // rotate acceleration
         Eigen::Vector3d acc(imu_in.linear_acceleration.x, imu_in.linear_acceleration.y, imu_in.linear_acceleration.z);
-        acc = extRot * acc;
+        acc = extRot * imuAcc.inverse() * acc;
```

Comment:
The correction matrices you intend to use would probably be strictly orthonormal, right? Using `.transpose()` instead of `.inverse()` would be a good idea here and below.

Reply:
`extRot` is orthonormal. The correction matrix I am incorporating (`imuAcc` in this snippet) is added because the IMU data isn't actually orthogonal due to manufacturing tolerances. That is to say that `imuAcc` accounts for the `psi`s in the following image:
![](https://www.analog.com/-/media/images/analog-dialogue/en/volume-49/number-2/articles/mems-imu-gyroscope-alignment/132108-fig-01.png?la=en)
I am aligning the IMU data (solid blue arrows) to  a perfectly orthogonal frame (dashed grey arrows). Therefore, since this correction matrix is not orthonormal, you cannot replace `.inverse()` with `.transpose()`.